source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -377,6 +377,38 @@ absl::optional<SystemTime> SslSocket::expirationPeerCertificate() const {
   return Utility::getExpirationTime(*cert);
 }
 
+bool SslSocket::peerCertificateChainPresented() const {
+  STACK_OF(X509)* cert_chain = SSL_get_peer_cert_chain(ssl_.get());
+  return cert_chain != nullptr && sk_X509_num(cert_chain) > 0;
+}
+
+const std::string& SslSocket::pemEncodedPeerCertificateChain() const {
```

Comment:
Note: if you want to use this in `X-Forwarded-Client-Cert` header, then this has to become `urlEncodedPemEncodedPeerCertificateChain()` in order to encode newline characters.

Reply:
Addressed in 626e635.